spec/services/events/stores/aggregated_clickhouse_store_spec.rb

```diff
@@ -0,0 +1,167 @@
+
+  let(:started_at) { Time.zone.parse("2023-03-15") }
+  let(:code) { billable_metric.code }
+
+  let(:boundaries) do
+    {
+      from_datetime: subscription.started_at.beginning_of_day,
+      to_datetime: subscription.started_at.end_of_month.end_of_day,
+      charges_duration: 31
+    }
+  end
+
+  let(:charge) { create(:standard_charge, plan:, billable_metric:) }
+
+  let(:grouped_by) { nil }
+  let(:grouped_by_values) { nil }
+  let(:with_grouped_by_values) { nil }
+
+  let(:charge_id) { charge.id }
+  let(:charge_filter) { nil }
+  let(:matching_filters) { {} }
+  let(:ignored_filters) { [] }
+
+  let(:events) do
+    events = []
+
+    5.times do |i|
+      properties = {billable_metric.field_name => i + 1}
+      groups = {}
+
+      if i.even?
+        # matching_filters.each { |key, values| properties[key] = values.first }
+
+        applied_grouped_by_values = grouped_by_values || with_grouped_by_values
+
+        if applied_grouped_by_values.present?
+        # applied_grouped_by_values.each { |grouped_by, value| properties[grouped_by] = value }
+        elsif grouped_by.present?
+          grouped_by.each do |group|
+            groups[group] = "#{Faker::Fantasy::Tolkien.character.delete("'")}_#{i}"
+          end
+        end
+      elsif grouped_by.present?
+        grouped_by.each do |group|
+          groups[group] = described_class::NIL_GROUP_VALUE
+        end
+      end
+
+      (ignored_filters.first || {}).each { |key, values| properties[key] = values.first } if i.zero?
+
+      events << Clickhouse::EventsEnrichedExpanded.create!(
+        transaction_id: SecureRandom.uuid,
+        organization_id: organization.id,
+        external_subscription_id: subscription.external_id,
+        subscription_id: subscription.id,
+        plan_id: plan.id,
+        code:,
+        aggregation_type: "count",
+        charge_id:,
+        charge_version: charge.updated_at,
+        charge_filter_id: charge_filter&.id,
+        charge_filter_version: charge_filter&.updated_at,
+        timestamp: boundaries[:from_datetime] + (i + 1).days,
+        properties:,
+        value: (i + 1).to_s,
+        decimal_value: (i + 1).to_d,
+        precise_total_amount_cents: i + 1,
+        grouped_by: groups
+      )
+    end
+
+    events
+  end
+
+  # NOTE: this does not include test with real values yet as we have to figure out
+  #       how to add factories of fixtures in spec env and to setup clickhouse on the CI
+  before do
+    if ENV["LAGO_CLICKHOUSE_ENABLED"].blank?
+      skip
+    else
+      events
+    end
+  end
+
+  after do
+    next if ENV["LAGO_CLICKHOUSE_ENABLED"].blank?
+
+    Clickhouse::BaseRecord.connection.execute("TRUNCATE TABLE events_enriched")
+    Clickhouse::BaseRecord.connection.execute("TRUNCATE TABLE events_enriched_expanded")
+    Clickhouse::BaseRecord.connection.execute("TRUNCATE TABLE events_aggregated")
+  end
+
+  describe ".count" do
+    it "returns the number of unique events" do
```

Comment:
Ain't we missing some usecase (`force_from`, `ordered`, `grouped_by_values`, etc.) here ?